variables.tf

```
@@ -471,8 +448,27 @@ variable "usage_limits" {
 
 variable "authentication_profiles" {
   description = "Map of authentication profiles to create"
-  type        = any
-  default     = {}
+  type = map(object({
+    name    = optional(string) # Will fall back to key if not set
+    content = any
+  }))
+  default  = {}
+  nullable = false
+}
+
+################################################################################
+# Logging
+################################################################################
+
+variable "logging" {
+  description = "Logging configuration for the cluster"
+  type = object({
+    bucket_name          = optional(string)
+    log_destination_type = optional(string)
+    log_exports          = optional(list(string), [])
```

This is leaving some perpetual drift:
```
  # module.redshift.aws_redshift_logging.this[0] will be updated in-place
  ~ resource "aws_redshift_logging" "this" {
        id                 = "ex-complete"
      + log_exports        = []
        # (4 unchanged attributes hidden)
    }
```
```suggestion
    log_exports          = optional(list(string))
```

---

updated in https://github.com/terraform-aws-modules/terraform-aws-redshift/pull/110/commits/2f27803a1088c3f5f250bbb8df8cb6999e26fd7c